A14-channelz.md

```diff
@@ -0,0 +1,1051 @@
+gRPC Channelz
+====
+* Author(s): Carl Mastrangelo
+* Approver: a11r
+* Status: In Review
+* Implemented in: 
+* Last updated: 11/1/17
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/5IYOMVm0Ufs
+
+Abstract
+----
+Adds channel level debug information that can be remotely queried.
+
+Background
+----
+
+gRPC needs a way to expose internal details about connections.  This
+information can be used to debug or diagnose a live program which may be
+exhibiting undesirable network behavior.  In Stubby, this need is fulfilled by
+the "rpcz" page, which exposes aggregated RPC stats as well as listing all
+incoming and outgoing connections.  gRPC would like to include and improve on
+this useful service.
+
+
+# Channelz Service
+
+The Channelz Service, hereafter channelz, reports all known channels, 
+subchannels, servers, and connections that process currently has.  In a process,
+there are multiple clients and servers.  To avoid confusion, a strict
+terminology will be used:
+
+1.  A "channel" represents an abstraction that can start and complete RPCs.  A
+    channel may have channels and subchannels.  A channel is a directed 
+    acyclic graph (DAG).  Only leaf channels may have
+    sockets.  Interior channels may only have channels and subchannels.
+2.  A "subchannel" represents an abstraction that is load balanced over by an 
+    owning channel. A subchannel may have channels and subchannels.  A 
+    subchannel is a directed acyclic graph (DAG).  Only leaf subchannels may 
+    have sockets.  Interior subchannels may only have channels and subchannels.
+3.  A descendent channel is either a channel or subchannel that is logically 
```

Comment:
Maybe put "descendent channel" in quotes, for consistency with the other definitions here?